dashboards/support.jsonnet

```diff
@@ -1,188 +1,205 @@
+      |||
+    )
+    + prometheus.withLegendFormat('{{ device }}'),
+  ]);
 
-local nfsServerReadLatency = graphPanel.new(
-  'NFS Server disk read latency',
-  min=0,
-  datasource='$PROMETHEUS_DS'
-).addTargets([
-  prometheus.target(
-    'sum(rate(node_disk_read_time_seconds_total{job="prometheus-nfsd-server"}[5m])) by (device) / sum(rate(node_disk_reads_completed_total{job="prometheus-nfsd-server"}[5m])) by (device)',
-    legendFormat='{{device}}'
-  ),
-]);
+local nfsServerReadLatency =
+  common.tsOptions
+  + ts.new('NFS Server disk read latency')
+  + ts.queryOptions.withTargets([
+    prometheus.new(
+      '$PROMETHEUS_DS',
+      |||
+        sum(rate(node_disk_read_time_seconds_total{job="prometheus-nfsd-server"}[5m])) by (device) / sum(rate(node_disk_reads_completed_total{job="prometheus-nfsd-server"}[5m])) by (device)
+      |||
+    )
+    + prometheus.withLegendFormat('{{ device }}'),
+  ]);
 
 // Support Metrics
-local prometheusMemory = graphPanel.new(
-  'Prometheus Memory (Working Set)',
-  formatY1='bytes',
-  min=0,
-  datasource='$PROMETHEUS_DS'
-).addTargets([
-  prometheus.target(
-    'sum(container_memory_working_set_bytes{pod=~"support-prometheus-server-.*", namespace="support"})'
-  ),
-]);
+local prometheusMemory =
+  common.tsOptions
+  + ts.new('Prometheus Memory (Working Set)')
+  + ts.standardOptions.withUnit('bytes')
+  + ts.queryOptions.withTargets([
+    prometheus.new(
+      '$PROMETHEUS_DS',
+      |||
+        sum(container_memory_working_set_bytes{pod=~"support-prometheus-server-.*", namespace="support"})
+      |||
+    ),
+  ]);
 
-local prometheusCPU = graphPanel.new(
-  'Prometheus CPU',
-  min=0,
-  datasource='$PROMETHEUS_DS'
-).addTargets([
-  prometheus.target(
-    'sum(rate(container_cpu_usage_seconds_total{pod=~"support-prometheus-server-.*",namespace="support"}[5m]))'
-  ),
-]);
+local prometheusCPU =
+  common.tsOptions
+  + ts.new('Prometheus CPU')
+  + ts.queryOptions.withTargets([
+    prometheus.new(
+      '$PROMETHEUS_DS',
+      |||
+        sum(rate(container_cpu_usage_seconds_total{pod=~"support-prometheus-server-.*",namespace="support"}[5m]))
+      |||
+    ),
+  ]);
 
-local prometheusDiskSpace = graphPanel.new(
-  'Prometheus Free Disk space',
-  formatY1='bytes',
-  min=0,
-  datasource='$PROMETHEUS_DS'
-).addTargets([
-  prometheus.target(
-    'sum(kubelet_volume_stats_available_bytes{namespace="support",persistentvolumeclaim="support-prometheus-server"})'
-  ),
-]);
+local prometheusDiskSpace =
+  common.tsOptions
+  + ts.new('Prometheus Free Disk space')
+  + ts.standardOptions.withUnit('bytes')
+  + ts.queryOptions.withTargets([
+    prometheus.new(
+      '$PROMETHEUS_DS',
+      |||
+        sum(kubelet_volume_stats_available_bytes{namespace="support",persistentvolumeclaim="support-prometheus-server"})
+      |||
+    ),
```

Comment:
I think we should add a legend here, otherwise the query is shown on the axis and it's confusing. This shouldn't block this PR though, as it wasn't there before either:
<img width="708" alt="Screenshot 2024-04-11 at 15 59 08" src="https://github.com/jupyterhub/grafana-dashboards/assets/7579677/91551b86-c4e8-4fc5-8d69-96d63a7b0ecb">

Prometheus cpu, memory  and disk all show the query as label.

Reply:
I opened https://github.com/jupyterhub/grafana-dashboards/issues/117 to track this